### PR TITLE
Update linux LLVM to 8, and don't build crystal twice

### PR DESCRIPTION
**Pull request by RX14**
I was doing this anyway, and it works on my machine, could you test if this fixes the CI?

### CODE REVIEW

File: linux/Makefile [hunk 1]
```diff
@@ -47,15 +47,15 @@ BUILD_ARGS_COMMON = $(DOCKER_BUILD_ARGS) \
 
 BUILD_ARGS64 = $(BUILD_ARGS_COMMON) \
                --build-arg previous_crystal_release=$(PREVIOUS_CRYSTAL_RELEASE_LINUX64_TARGZ)	\
-               --build-arg debian_image=debian:8 \
-               --build-arg alpine_image=alpine:3.8 \
+               --build-arg debian_image=debian:10 \
+               --build-arg alpine_image=alpine:edge \
```
Comment (bcardiff) [inline linux/Makefile]: We need specific tags here for better reproducibility. `:edge` is not an option.

File: linux/Makefile [hunk 2]
```diff
@@ -47,15 +47,15 @@ BUILD_ARGS_COMMON = $(DOCKER_BUILD_ARGS) \
...
                --build-arg musl_target=x86_64-linux-musl \
                --build-arg gnu_target=x86_64-unknown-linux-gnu
 
 BUILD_ARGS32 = $(BUILD_ARGS_COMMON) \
                --build-arg previous_crystal_release=$(PREVIOUS_CRYSTAL_RELEASE_LINUX32_TARGZ) \
-               --build-arg debian_image=i386/debian:8 \
-               --build-arg alpine_image=i386/alpine:3.8 \
+               --build-arg debian_image=i386/debian:10 \
+               --build-arg alpine_image=i386/alpine:edge \
```
Comment (bcardiff) [inline linux/Makefile]: ditto

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,11 +1,12 @@
-FROM ubuntu:xenial as runtime
+FROM ubuntu:bionic as runtime
```
Comment (j8r) [inline docker/Dockerfile]: Debian's `buster-slim` image would be more appropriate for Docker, slimmer.

Comment (bcardiff) [reply docker/Dockerfile]: Both are 27mb

Comment (j8r) [reply docker/Dockerfile]: Right, it has changed :+1: 
I remembered slim images was lighter.
So, as you wish then.

File: linux/Makefile (hunk 1 above)
Comment (j8r) [reply linux/Makefile]: Alpine `3.10`

File: docker/Dockerfile (hunk 3 above)
Comment (bcardiff) [reply docker/Dockerfile]: Yeah. ubuntu images got far better in the last couple of releases. Also, this images need to be aligned with crystal-dist repo were tagged releases are done.

I will actually rollback this change. It needs to be aligned with crystal-dist images for 64 and 32 bits and that means definitely updating to llvm 6 or 8 in linux. I would rather update them after some testing.

@RX14 let's keep xenial here for a little bit longer and upgrade in another PR

File: linux/Makefile (hunk 1 above)
Comment (bcardiff) [reply linux/Makefile]: edge is not 3.10 from docker hub. Was there an explicit need for edge @RX14?

File: docker/Makefile [hunk 4]
```diff
@@ -1,7 +1,7 @@
 CRYSTAL_VERSION ?=   ## How the binaries should be branded
 CRYSTAL_DEB ?=       ## which crystal.deb file to install in the docker images
-DOCKER_TAG ?=        ## How to tag the docker image. a -build will be appended for -build image
-DOCKER_REPOSITORY ?= ## Docker hub repository to commit image
+DOCKER_TAG ?= latest## How to tag the docker image. a -build will be appended for -build image
```
[the remark this answers is not in the document]
Comment (RX14) [reply docker/Makefile]: Yes, they should. They're good defaults for local builds. The CI will override them. This change does not change any behaviour that would not already fail.

File: linux/Makefile (hunk 1 above)
Comment (RX14) [reply linux/Makefile]: Yes, `ldd` is broken on 3.10, see https://github.com/crystal-lang/crystal/issues/7946.

We can switch to 3.11 when it's released, depending on `edge` is fine.

File: docker/Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (bcardiff) [reply docker/Makefile]: The story of a local build is half way.

To support local builds a more complete solution would be to have a script that will fetch latest release and master's sha1 for example.

But all this is independant of the intention of the PR to update llvm 8 and avoid building crystal twice.

File: docker/Dockerfile (hunk 3 above)
Comment (RX14) [reply docker/Dockerfile]: If you minsist.

File: docker/Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (RX14) [reply docker/Makefile]: It's a harmless change. I'm not going to revert it.

**Comment by RX14** [comment]
Closes #29 btw

File: linux/Makefile (hunk 1 above)
Comment (bcardiff) [reply linux/Makefile]: We can't depend on edge. Using the distribution-scripts in the future will not be as reproducible as using a specific version of the docker image.

Maybe we need to delay this until 3.11 is released and stick with #43

Comment (RX14) [reply linux/Makefile]: *sigh*, I'll just patch the ldd script manually

**Comment by RX14** [comment]
okay, there we go

sorry for the force push, accidentallied

**Comment by bcardiff** [comment]
Cool thanks.
Checking the build at https://circleci.com/workflow-run/35cfe1aa-4ce3-466b-a8c4-49a77e4ef610

**Pull request merged**